sdk/core/azure-core/src/main/java/com/azure/core/implementation/RestProxy.java

```diff
@@ -154,6 +164,78 @@ public Object invoke(Object proxy, final Method method, Object[] args) {
+                                currentTotalLength, expectedLength), currentTotalLength, expectedLength);
+                        }
+                        bufferLengthList.add(s.get().remaining());
+                    } else if (s.isOnComplete()) {
+                        Long currentTotalLength = Long.valueOf(
+                            bufferLengthList.stream().reduce(Integer::sum).orElse(0));
+                        if (expectedLength.compareTo(currentTotalLength) != 0) {
+                            throw new UnexpectedLengthException(
+                            String.format("Request body emitted %d bytes less than the expected %d bytes.",
+                                currentTotalLength, expectedLength), currentTotalLength, expectedLength);
+                        }
+                    } else {
+                        logger.logExceptionAsError(new RuntimeException("Error occurs when validating "
+                            + "the request body legnth and the header content length. Error details: "
+                            + s.getThrowable().getMessage()));
+                    }
+                });
+            }
+        });
+    }
+
+//    private Flux<ByteBuffer> validateLength(final HttpRequest request, final Flux<ByteBuffer> bbFlux) {
+//        if (bbFlux == null) {
+//            return Flux.empty();
+//        }
+//        return Flux.defer(new Supplier<Publisher<ByteBuffer>>() {
+//            @Override
+//            public Publisher<ByteBuffer> get() {
+//                Long expectedLength = Long.valueOf(request.headers().value("Content-Length"));
+//                Mono<byte[]> bbMono = FluxUtil.collectBytesInByteBufferStream(bbFlux).doOnNext(actualLength -> {
+//                    if (actualLength.length > expectedLength) {
+//                        throw new UnexpectedLengthException(
+//                            String.format("Request body emitted more bytes than the expected %d bytes.",
+//                                expectedLength), actualLength.length, expectedLength);
+//                    } else if (actualLength.length < expectedLength) {
+//                        throw new UnexpectedLengthException(
+//                            String.format("Request body emitted less bytes than the expected %d bytes.",
+//                                expectedLength), actualLength.length, expectedLength);
+//                    }
+//                });
+//
+//                return bbMono.flatMapMany(new Function<byte[], Publisher<ByteBuffer>>() {
+//                    @Override
+//                    public Publisher<ByteBuffer> apply(final byte[] bytes) {
+//                        return Flux.just(ByteBuffer.wrap(bytes));
+//                    }
+//                });
+//            }
+//        });
+//    }
+
+    private int len(ByteBuffer input) {
+        int result = input.remaining();
```

Comment:
Is this method necessary? Can't we just use `ByteBuffer::remaining` or `input.remaining()` wherever this is being called?

Reply:
Removed.